### ISSUE TITLE
Bookmarks sorting improvements

### ISSUE THREAD

**Issue by thany** [organicmaps/organicmaps#4250]
I tried sorting bookmarks. Not the list of bookmark lists, but a single list of bookmarks. I found a couple of things that need improving:

* **Sorting by title** - Usecase is my list of hotels to visit on my upcoming citytrip. I was thinking to prepend them by a number, and then sorting them by name, so that they appear in order of visiting them.
* **Custom sorting** - This might be achieved by tap-and-holding a bookmark in the list, and then dragging it to whereever it need to go.
* **Rename "default" order** - It's unclear what "sort by default" means. Probably order in which bookmarks got added or something, which is fine. Anyways, please just call it what it actually is.
* **Reverse sort order** - Just to order ascending/descending. Probably shouldn't be available for custom sort (or should it? I dunno, maybe it could work).

The other options should stay as well. I can see the usecase for them.

And maybe one bonus improvement: please add a way to change the bookmark icon. Currently it is derived from the POI type on OSM, which is a perfectly good default icon. But when I add hotels on there that are not yet on OSM (I added them, don't worry) they show up using a generic icon. It'd be neat if I can tell the app what kind of POI a bookmark is. The relevance to sorting is that this will make "Sort by type" work more correctly.

**Comment by biodranik** [comment]
There is also #1066 and likely some other related issues.

**Comment by thany** [comment]
Yes, I just thought I'd create an issue that paints the whole picture of needed bookmarks improvements (the way I see it anyway).

**Comment by ioctl-user** [comment]
Some additional suggestions:

-  **Sorting by tags** -- add tags to bookmarks. Can be used select bookmarks for one country, for example.
- **Implement bookmarks directories**. Can be used to organize bookmarks, for example, by creating directory for each continent or country.

**Comment by thany** [comment]
@ioctl-user Those are whole new features. Best to create separate issues for them.

**Comment by ioctl-user** [comment]
> @ioctl-user Those are whole new features. Best to create separate issues for them.

Done:  https://github.com/organicmaps/organicmaps/issues/4381

**Comment by oleg-rswll** [comment]
Related to #5680
